client/history/client.go

```diff
@@ -820,9 +821,10 @@ func (c *clientImpl) GetReplicationMessages(
 			if err != nil {
 				c.logger.Warn("Failed to get replication tasks from client", tag.Error(err))
 				// Returns service busy error to notify replication
-				if _, ok := err.(*types.ServiceBusyError); ok {
+				var e *types.ServiceBusyError
+				if errors.As(err, &e) {
 					select {
-					case errChan <- err:
+					case errChan <- e:
```

Comment:
this... might be the desired behaviour? I figure if it 's a wrapped error, the channel won't expect having a wrapper being passed along, but it does mean the wrap is lost in this instance